R/helper_item_deletion.R

```diff
@@ -0,0 +1,341 @@
+#' and returns two restructured data frames with the classification accuracy   
+#' indices for the reference and focal groups under strict invariance and 
+#' partial invariance conditions, and the corresponding h for the difference in  
+#' CAI between the two invariance conditions for each group.
+#' @param strict_output Output from [PartInv()] under strict invariance. 
+#' @param partial_output Output from [PartInv()] under partial invariance. 
+#' @return A 8 x 3 dataframe with columns `strict invariance`, 
+#'        `partial invariance`, and `h`.
+acc_indices_h <- function(strict_output, partial_output) {
+  r_names <- c("TP", "FP", "TN", "FN", "PS", "SR", "SE", "SP")
+  
+  ref_par_strict <- partial_output$summary[1][, 1]
+  ref_strict <- strict_output$summary[1][, 1]
+  
+  df_ref <- data.frame(SFI =  ref_strict, 
+                   PFI = ref_par_strict, row.names = r_names)
+  df_ref["h"] <- cohens_h(df_ref$SFI, df_ref$PFI)
+  
+  f_par_strict <- partial_output$summary[2][, 1]
+  f_strict <- strict_output$summary[2][, 1]
+  df_f <- data.frame(SFI =  f_strict, 
+                       PFI = f_par_strict, row.names = r_names)
+  df_f["h"] <- cohens_h(df_f$SFI, df_f$PFI)
+  return(list("Reference" = df_ref, "Focal" = df_f))
+}
+ 
+
+#' @title 
+#' Determine biased items
+#' 
+#' @name 
+#' determine_biased_items
+#' 
+#' @description 
+#' \code{determine_biased_items} takes in the factor loadings, intercepts, and
+#'  uniqueness for the reference and focal groups, and returns indices of 
+#'  noninvariant items.
+#'  
+#' @param lambda_r Factor loadings for the reference group.
+#' @param lambda_f Factor loadings for the focal group.
+#' @param nu_r Measurement intercepts for the reference group.
+#' @param nu_f Measurement intercepts for the focal group.
+#' @param Theta_r Uniqueness for the reference group.
+#' @param Theta_f Uniqueness for the focal group.
+#' @param weights Vector of item weights. If an item weight is assigned to be 0, 
+#' it won't be returned as a biased item.
+#' 
+#' 
+#' @return A vector containing the indices of the biased items.
+#' @examples 
+#' lambda_matrix <- matrix(0, nrow = 5, ncol = 2)
+#' lambda_matrix[1:2, 1] <- c(.322, .655)
+#' lambda_matrix[3:5, 2] <- c(.398, .745, .543)
+#' determine_biased_items(lambda_r = lambda_matrix,
+#'                        lambda_f = lambda_matrix, 
+#'                        nu_r = c(.225, .025, .010, .240, .125),
+#'                        nu_f = c(.225, -.05, .240, -.025, .125),
+#'                        Theta_r = diag(1, 5),  
+#'                        Theta_f = diag(c(1, .95, .80, .75, 1)), 
+#'                        weights = c(1/4, 1/4, 1/6, 1/6, 1/6))       
+#' @export
+determine_biased_items <- function(lambda_r, lambda_f, nu_r, nu_f, 
+                                   Theta_r, Theta_f, weights) {
+  biased_lambda <- biased_theta <- biased_nu <- c()
+  
+  # Compare factor loadings 
+  if(is.matrix(lambda_r)) {
+    for(i in seq_len(ncol(lambda_r))) {
```

Comment:
Here's an alternative:

```r
which(!apply(lambda_r == lambda_f, 1, all))
```

Reply:
And you can convert the input to matrix with `as.matrix(lambda_r)` to avoid if-else, although that may not be the most memory efficient.